eureka-client/src/main/java/com/netflix/discovery/shared/resolver/EurekaEndpoint.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.discovery.shared.resolver;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Tomasz Bak
+ */
+public class EurekaEndpoint implements Comparable<EurekaEndpoint> {
```

Comment:
On a second thought, I am not sure about that. In ClusterResolver we have 'region' attribute. This is useful metadata, and it is convenient to carry it together with the ClusterResolver object, just like zone with EndpointResolver. In Eureka server we will have multiple copies of ClusterResolver, one for local, and a few for remote regions. We could provide AWS specific abstraction here as well, but in this case region name serves as an id.
Other cloud providers use similar terms (Google Compute Engine region/zone, Azure/RackSpace region). At least 'region' seems to be a common one.